python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/antlr_asset_selection.py

```diff
@@ -121,25 +124,35 @@ def visitKeyExpr(self, ctx: AssetSelectionParser.KeyExprContext):
     def visitTagAttributeExpr(self, ctx: AssetSelectionParser.TagAttributeExprContext):
         key = self.visit(ctx.value(0))
         value = self.visit(ctx.value(1)) if ctx.EQUAL() else ""
-        return AssetSelection.tag(key, value, include_sources=self.include_sources)
+        return AssetSelection.tag(
+            key, value, include_sources=self.include_sources, is_null=is_null_value(ctx.value(0))
+        )
 
     def visitOwnerAttributeExpr(self, ctx: AssetSelectionParser.OwnerAttributeExprContext):
         owner = self.visit(ctx.value())
-        return AssetSelection.owner(owner)
+        return AssetSelection.owner(owner, is_null=is_null_value(ctx.value()))
 
     def visitGroupAttributeExpr(self, ctx: AssetSelectionParser.GroupAttributeExprContext):
         group = self.visit(ctx.value())
-        return AssetSelection.groups(group, include_sources=self.include_sources)
+        return AssetSelection.groups(
+            group, include_sources=self.include_sources, is_null=is_null_value(ctx.value())
+        )
 
     def visitKindAttributeExpr(self, ctx: AssetSelectionParser.KindAttributeExprContext):
         kind = self.visit(ctx.value())
-        return AssetSelection.tag(f"{KIND_PREFIX}{kind}", "", include_sources=self.include_sources)
+        return AssetSelection.kind(
+            kind,
+            include_sources=self.include_sources,
+            is_null=is_null_value(ctx.value()),
+        )
```

Comment:
I added an `AssetSelection.kind` instead of overloading `AssetSelection.tag` because otherwise the toString and back doesn't actually work correctly...

Reply:
Ah yeah so currently catalog views stores the raw string. if we switched to storing the serdes object (which we want to do long term so that we don't need to migrate strings) then we'd have the issue